client/cli/src/runtime.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use futures::{Future, future, future::FutureExt};
+use futures::select;
+use futures::pin_mut;
+use sc_service::{AbstractService, Configuration};
+use crate::error;
+use crate::informant;
+
+async fn main<F, E>(func: F) -> Result<(), Box<dyn std::error::Error>>
+where
+	F: Future<Output = Result<(), E>> + future::FusedFuture,
+	E: 'static + std::error::Error,
+{
+	use tokio::signal::unix::{signal, SignalKind};
```

Comment:
Using this would mean that substrate wouldn't compile on windows, but I'm not sure we care about that.

Reply:
I thought it was not supported but now that I checked the doc, they do mention building substrate entirely on Windows.

Reply:
I just pushed a fixed for that